### PR TITLE
add resolved IPs and CNAMEs to DNS report

### PR DESCRIPTION
**Pull request by philipatl**
## Description

Enforcer should make DNS reports more meaningful. Enforcer should report the resolved IPs (and any CNAMEs). Enforcer should also report whether a DNS request failed.

## Motivation and Context

We would like to improve our DNS reports and make them meaningful. There are also other improvements mentioned in CNS-1715 that this PR will lay the groundwork for.

## How Has This Been Tested?

More backend work is required before it can be tested end-to-end, but enforcer unit tests have been added.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "082c5931aca702a664b0086524351c721e5f59c2",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "23a2a178db385ddc38d957f33385f436a9827a0c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "23a2a178db385ddc38d957f33385f436a9827a0c",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "5e84f716609f7bc085b3338f5633f347e112960b",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "244202d20ec3a14e612af51bdd4be4060f8f8022",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "5106e3d191c241fadbed79ea26d1be5cdffe30ef",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "244202d20ec3a14e612af51bdd4be4060f8f8022",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "76bcef3c5a87dd098a8c6b83cc9bce14afefdad5",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "244202d20ec3a14e612af51bdd4be4060f8f8022",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "76bcef3c5a87dd098a8c6b83cc9bce14afefdad5",
    "pipeline": "master"
  }
]
```

File: specs/dnslookupreport.spec [hunk 1]
```diff
@@ -98,6 +100,26 @@ attributes:
...
+      bson_name: g
+
+  - name: resolvedIPs
+    description: resolved IP addresses.
+    type: list
+    exposed: true
+    subtype: string
+    stored: true
+    omit_empty: true
+    extensions:
```
Comment (primalmotion) [inline specs/dnslookupreport.spec:120]: needs to use the custom validation for IP lists

File: specs/dnslookupreport.spec [hunk 2]
```diff
@@ -107,7 +129,7 @@ attributes:
     example_value: www.google.com
     omit_empty: true
     extensions:
-      bson_name: g
+      bson_name: i
```
Comment (primalmotion) [inline specs/dnslookupreport.spec:132]: this completely breaks backward compatibility. Please never edit bson_name if they've already been set.

Comment (philipatl) [reply specs/dnslookupreport.spec]: Thanks, I've fixed this. The two new properties also have unique and sequential `bson_name` values.

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "30d3102289a7c136891219abc24dcf08e7b090a9",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "9169bb8085a3b440b9472899d75336e80764c7f2",
    "pipeline": "master"
  }
]
```

**Review by primalmotion — APPROVED** [review]
LGTM

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "9169bb8085a3b440b9472899d75336e80764c7f2",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "30d3102289a7c136891219abc24dcf08e7b090a9",
    "pipeline": "master"
  }
]
```

**Comment by itsai220** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "30d3102289a7c136891219abc24dcf08e7b090a9",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "9169bb8085a3b440b9472899d75336e80764c7f2",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "1bd213355d992cf7e5b22d0765f432f111635697",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "1bd213355d992cf7e5b22d0765f432f111635697",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "9169bb8085a3b440b9472899d75336e80764c7f2",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "30d3102289a7c136891219abc24dcf08e7b090a9",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "f25fafb9dcea11e39a7faa61d004b157ab719f20",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "9169bb8085a3b440b9472899d75336e80764c7f2",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "f25fafb9dcea11e39a7faa61d004b157ab719f20",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "a0d1a9bd1e31a24b251b8f16561e78a86f48427a",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "a0d1a9bd1e31a24b251b8f16561e78a86f48427a",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "ab0cd0ef97c2d16ad4f098eb445e960f2f8b38e9",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "ab0cd0ef97c2d16ad4f098eb445e960f2f8b38e9",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "67efde07d7c90d994c503f992f8cd32d55cf133c",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "e2f4b6d0cb04565c195aca8857a37f373a4d8d1d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "eabf1d3289c559408aef2a3d128e4d9b9e0e3749",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "e2f4b6d0cb04565c195aca8857a37f373a4d8d1d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "eabf1d3289c559408aef2a3d128e4d9b9e0e3749",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "7ea6332335abcde22bf7c0ef07052d854dee5ab9",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "463bb39c680a4ce42914de84e4acb71708b4d94e",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "e2f4b6d0cb04565c195aca8857a37f373a4d8d1d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "eabf1d3289c559408aef2a3d128e4d9b9e0e3749",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "b7283cbaf08ec8fd9ec1656808b2d74b3b65bff0",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "993fc325db170f1342a45007fc9d4ded067ac481",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "463bb39c680a4ce42914de84e4acb71708b4d94e",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "5f7fe7754536b7dc485d2bd4b46d72c73d45099c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "b7283cbaf08ec8fd9ec1656808b2d74b3b65bff0",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "b2b1fdbdfc0df98f09cf7c03389e2cdc90d82b8d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "b2ee4874a650774ef1fa60d62514f62cff169827",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "5f7fe7754536b7dc485d2bd4b46d72c73d45099c",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "b7283cbaf08ec8fd9ec1656808b2d74b3b65bff0",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "b2b1fdbdfc0df98f09cf7c03389e2cdc90d82b8d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "b2ee4874a650774ef1fa60d62514f62cff169827",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "5f7fe7754536b7dc485d2bd4b46d72c73d45099c",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests-lib",
    "pr-id": "510",
    "commit-sha": "95f3aaf5ed9cafb03e829e89eaadc4bb7360abdf",
    "pipeline": "master"
  }
]
```

**Comment by philipatl** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "backend",
    "pr-id": "1148",
    "commit-sha": "b7283cbaf08ec8fd9ec1656808b2d74b3b65bff0",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "gaia",
    "pr-id": "635",
    "commit-sha": "b2b1fdbdfc0df98f09cf7c03389e2cdc90d82b8d",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "enforcerd",
    "pr-id": "2370",
    "commit-sha": "b2ee4874a650774ef1fa60d62514f62cff169827",
    "pipeline": "master"
  },
  {
    "project": "dns-report-resolved-ips-NEW",
    "component": "apotests",
    "pr-id": "3321",
    "commit-sha": "cb2c32ab4ef0d0dd199390acdc1a82d2ae5dcd6e",
    "pipeline": "master"
  }
]
```

**Pull request merged**